modules/axtask/src/task.rs

```diff
@@ -398,6 +404,22 @@ impl TaskInner {
         self.ctx.get()
     }
 
+    /// Returns the CPU ID where the task is running or will run.
+    ///
+    /// Note: the task may not be running on the CPU, it just exists in the run queue.
+    #[cfg(feature = "smp")]
+    #[inline]
+    pub fn cpu_id(&self) -> u32 {
+        self.cpu_id.load(Ordering::Acquire)
+    }
+
+    /// Set the CPU ID where the task is running or will run.
+    #[cfg(feature = "smp")]
+    #[inline]
+    pub fn set_cpu_id(&self, cpu_id: u32) {
```

Comment:
Others should not modify the CPU ID of a task, make this function `pub(crate) `.